topology/topology.go

```diff
@@ -136,9 +137,26 @@ func (topology *TopologySvc) handleHostListGet(input interface{}, ctx common.Res
 	return retval, nil
 }
 
+// handleHostListPost handles addition of a host to the current datacenter.
+// If the HostMessage.AgentPort is not specified, root service is queried for
+// the default Agent port.
 func (topology *TopologySvc) handleHostListPost(input interface{}, ctx common.RestContext) (interface{}, error) {
 	hostMessage := input.(*common.HostMessage)
-	host := Host{Ip: hostMessage.Ip, Name: hostMessage.Name, RomanaIp: hostMessage.RomanaIp, AgentPort: uint64(hostMessage.AgentPort)}
+	var port uint64
+	if hostMessage.AgentPort == 0 {
```

Comment:
I don't know what's happening here. Could you add some comments for explanation, please?